### ISSUE TITLE
insert signature at point?

### ISSUE THREAD

**Issue by brittAnderson** [hexmode/gnus-alias#5]
Occasionally I edit a message where I want to reply to a fragment of the message inline, but leave the remaining text after my response as a reference for the original sender (usually someone who only knows about top posting). I would like to sign the message where my interleaved replies end. It would be nice to have the signature that gets inserted when I switch identities appear at that point rather than at the bottom of the message. Now I just go to the bottom, grab the signature block and return to my comment and paste it. 

Is there a way to achieve this behavior with the current package? Or if not, I would be happy to use it as an occasion to try and learn a little elisp if you would like to point me to the approach that you think would most efficiently achieve this with the current package design. 

Thanks.

**Comment by hexmode** [comment]
I don't think this currently exists.  However a quick look at the source shows [`gnus-alias-remove-sig`](https://github.com/hexmode/gnus-alias/blob/master/gnus-alias.el#L1263) which just deletes from the `message-signature-separator` (usually `-- `) to the end of the buffer.

You could copy that and then insert your current sig a point using something like the following (untested):
``` elisp
(let ((current (point)))
  (gnus-alias-goto-sig)
  (kill-region (point) (point-max))
  (goto-char current)
  (yank-pop))

**Comment by hexmode** [comment]
Here is something that is tested, but works only once.  You can't switch the alias after using it, for example:
``` elisp
(let ((current (point)))
  (when (re-search-forward message-signature-separator nil 'move)
    (beginning-of-line 0))
  (kill-region (point) (point-max))
  (goto-char current)
  (insert-for-yank (current-kill 0))
  (goto-char current))

**Comment by brittAnderson** [comment]
Thanks for this. This works great. I have added it to my init file when gnus-alias loads and all is good. Thanks. Please feel free to close.
